### PR TITLE
[PSA-2298] bump kube-rbac-proxy to 0.18.1

### PR DESCRIPTION
**Pull request by mcollao-hc**
Bumping kube-rbac-proxy to 0.18.1 which solves multiple vulnerabilities.

https://github.com/brancz/kube-rbac-proxy/releases/tag/v0.18.1

### CODE REVIEW

File: config/default/manager_auth_proxy_patch.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ spec:
         # capabilities:
         #   drop:
         #     - "ALL"
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.15.0
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.18.1
```
Comment (benashz) [inline config/default/manager_auth_proxy_patch.yaml:22]: That should be:
```suggestion
        image: quay.io/brancz/kube-rbac-proxy:v0.18.1
```

File: chart/values.yaml [hunk 2]
```diff
@@ -121,7 +121,7 @@ controller:
     image:
       pullPolicy: IfNotPresent
       repository: gcr.io/kubebuilder/kube-rbac-proxy
```
Comment (benashz) [inline chart/values.yaml:123]:
```suggestion
      repository: quay.io/brancz/kube-rbac-proxy
```

**Review by benashz — COMMENTED** [review]
Thanks for the contribution. It looks like we also need to update the repository. The tag you have specified does not exist on `gcr.io`. I have provided some suggested updates to address that.

We also need to to update the version here: https://github.com/hashicorp/vault-secrets-operator/blob/649d7cff7f18541da6c673c46b6a72362fc152e4/Makefile#L7

File: config/default/manager_auth_proxy_patch.yaml (hunk 1 above)
Comment (mcollao-hc) [reply config/default/manager_auth_proxy_patch.yaml]: Changed, thanks!

File: chart/values.yaml (hunk 2 above)
Comment (mcollao-hc) [reply chart/values.yaml]: Changed, thanks!

**Review by benashz — APPROVED** [review]
LGTM!

**Pull request merged**